concent_api/verifier/decorators.py

```diff
@@ -24,3 +28,38 @@ def wrapper(*args, **kwargs):
             # Remove any files left in VERIFIER_STORAGE_PATH.
             clean_directory(settings.VERIFIER_STORAGE_PATH)
     return wrapper
+
+
+def ensure_retry_of_locked_calls(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        repeat_handler = RepeatHandler()
+        wait_time_in_sec = 0.05
+        should_repeat = True
+        result = None
+
+        while should_repeat:
+            try:
+                result = func(*args, **kwargs)
+                should_repeat = False
+            except DatabaseError:
```

Comment:
Are you absolutely sure that `DatabaseError` always indicates a failure due to a timed out lock? Because I think that there are a lot more things that can cause this error and blindly retrying when they happen might actually be harmful.

We want to retry only if it's a problem with the lock. If there's a different problem we want to raise an exception and exit immediately.